### ISSUE TITLE
How are the input files generated?

### ISSUE THREAD

**Issue by JamieBurge-LT** [yaniker/one-shot-skill-assessment-in-high-stakes-domains-with-limited-data-via-meta-learning#1]
Hi @yaniker ,
I've been playing with the code recently and have everything running, I'm looking to use the model on our own data but couldn't find any mention of how the input .pt files are generated.
Could you please clarify:

What data or preprocessing steps were used to create these .pt files?
Is there a script or pipeline available in the repo (or elsewhere) that was used to generate them?
If not, could you provide a brief overview of the process?
This would really help in adapting the project to new data. Thanks in advance for your time and help!
Best regards,
Jamie Burge

**Comment by yaniker** [comment]
Hi Jamie,

I would be happy to help you with your questions:  

1. What data or preprocessing steps were used to create these .pt files?
```bash
The .pt files were generated using the SimCLR contrastive learning algorithm. You can find the details here: https://proceedings.mlr.press/v119/chen20j.html
```

2. Is there a script or pipeline available in the repo (or elsewhere) that was used to generate them? If not, could you provide a brief overview of the process?
```bash
Please refer to https://lightning.ai/docs/pytorch/stable/notebooks/course_UvA-DL/13-contrastive-learning.html
```

Please let me know if you have further questions.

Best,
Erim

**Comment by razvanilin** [comment]
Hi @yaniker,

Thanks again for pointing us to the SimCLR paper and the Lightning notebook—very helpful!  
After inspecting several of the sample `.pt` files, we still need a few concrete details so we can reproduce the exact feature tensors your model expects.

Could you please clarify (or share) the following?

1. **Script or pipeline**  
   - Do you have a script that converts raw videos into the `.pt` feature tensors? That would be the most helpful asset, as it would remove all ambiguity about sampling, transforms, and model setup.

2. **Checkpoint**  
   - If the script references a trained SimCLR model, could you share the weights (encoder + projection head) or a download link?

3. **Backbone & projection head**  
   - What base architecture did you use (e.g., ResNet-18, ResNet-50)?  
   - What is the projection-head output size? (We’ve seen tensors shaped like `(8, 1, 509)`.)

4. **Frame sampling strategy**  
   - What FPS or method was used to extract frames?  
   - Are sequences padded or trimmed to a fixed number of frames, or do the `.pt` files have variable length depending on the video? (We’ve seen the time dimension vary, such as `(8, 1, 509)`.)

5. **Pre-processing transforms**  
   - What image transforms were applied before feeding frames to the encoder? (Resize? Crop? RGB conversion? Normalization values?)

6. **Tensor contents**  
   - Is the final dimension (e.g., 509) the raw projection-head output per frame, or is it processed in any way (e.g., pooled, concatenated)?

Once we have these details—or ideally just the script—we’ll be able to generate our own `.pt` files that plug directly into the existing pipeline.

Really appreciate your time and help!

**Comment by razvanilin** [comment]
Hi @yaniker,

Thanks for the helpful references! We've made significant progress implementing our own SimCLR pipeline based on the paper, but we're still seeing suboptimal results and suspect some methodology differences.

**Our current implementation:**
- ResNet34 backbone with SimCLR pretraining 
- 1 FPS sampling, Global Average Pooling (512→8 dims)
- Output format: (8, 1, T) tensors matching your samples
- ~63% accuracy with 1200+ videos of Intracorporeal Suture

**We've discovered some discrepancies** between the paper's description and your actual implementation:
- Paper describes (T, D') tensor format, but your working model expects (D', 1, T)
- Paper mentions min-max normalization, but your samples seem to preserve natural feature distributions
- Other potential differences in preprocessing/training methodology

**Specific questions:**
1. **Could you share your actual SimCLR training script?** This would resolve any remaining implementation differences.

2. **Feature extraction details:**
   - Do you apply any normalization after GAP? (min-max, z-score, etc.)
   - Exact preprocessing transforms used on video frames?
   - Do you use the projection head output or just the backbone encoder?

3. **Training hyperparameters:**
   - Batch size, learning rate, temperature, epochs used?
   - Any specific augmentation strategies?

We'd really appreciate any code or additional implementation details you could share, as the general SimCLR references leave room for interpretation.

Thanks again for your time!
